core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1704,9 +1734,26 @@ object Log {
     * Return a directory name to rename the log directory to for async deletion. The name will be in the following
     * format: topic-partition.uniqueId-delete where topic, partition and uniqueId are variables.
     */
-  def logDeleteDirName(logName: String): String = {
+  def logDeleteDirName(topicPartition: TopicPartition): String = {
+    val uniqueId = java.util.UUID.randomUUID.toString.replaceAll("-", "")
```

Comment:
Seems like we should extract a method that just takes the suffix and generates the `uniqueId`, etc.

Reply:
Sure. I have updated the patch to add a private method `logDirName(topicPartition: TopicPartition, suffix: String)`. I still keep the method `logFutureDirName` and `logDeleteDirName` because it seems easy to use by the caller. I can remove these two methods if you prefer.